aws-s3-transfer-manager/src/operation/download.rs

```diff
@@ -44,6 +45,7 @@ impl Download {
     pub(crate) async fn orchestrate(
         handle: Arc<crate::client::Handle>,
         input: crate::operation::download::DownloadInput,
+        existing_span_for_tasks: Option<tracing::span::Id>,
```

Comment:
Why do we need to pass this in? Span relationships should generally be automatic.

Reply:
It's not automatic for spawned tasks, which start with no parent span. Previously, the "download-chunk" spans had no parent. If your main thread did 2 downloads, where each download had 2 chunks, the span tree would look like:
```
- main
  - initiate-download
  - join-download
  - initiate-download
  - join-download
- download-chunk
- download-chunk
- download-chunk 
- download-chunk <-- which "download operation" do these chunks belong to?
```

So I started creating "download-tasks" span that's used to group a download's spawned tasks together. So now it looks like:
```
- main
  - initiate-download
  - join-download
  - initiate-download
  - join-download
- download-tasks <-- yay a common parent per download operation
  - download-chunk
  - download-chunk
- download-tasks
  - download-chunk
  - download-chunk
```

But `download-objects` is going to do many downloads, which will lead to many different "download-tasks" spans, and I thought it would be nice to group them under a common parent.under another span. So when `download-objects` calls this, it passes a span named "download-objects-tasks" and the span tree looks like:
```
- main
  - initiate-download-objects
  - join-download-objects
- download-objects-tasks <-- 
  - download-tasks
    - download-chunk
    - download-chunk
  - download-tasks
    - download-chunk
    - download-chunk
```

Reply:
Shouldn't calling `.instrument()` on the future passed to `spawn()` be sufficient?